extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlCreate.java

```diff
@@ -0,0 +1,82 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.kubectl;
+
+import com.google.common.base.Strings;
+import io.kubernetes.client.Discovery;
+import io.kubernetes.client.common.KubernetesListObject;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.util.ModelMapper;
+import io.kubernetes.client.util.Namespaces;
+import io.kubernetes.client.util.generic.GenericKubernetesApi;
+import io.kubernetes.client.util.generic.KubernetesApiResponse;
+import io.kubernetes.client.util.generic.options.CreateOptions;
+import java.util.Optional;
+import java.util.Set;
+
+public class KubectlCreate extends Kubectl.ApiClientBuilder<KubectlCreate>
+    implements Kubectl.Executable<KubernetesObject> {
+
+  KubectlCreate() {}
+
+  private KubernetesObject targetObj;
+
+  public KubectlCreate load(KubernetesObject obj) {
+    this.targetObj = obj;
+    return this;
+  }
+
+  @Override
+  public KubernetesObject execute() throws KubectlException {
+    Discovery discovery = new Discovery(apiClient);
+    try {
+      Set<Discovery.APIResource> apiResources = discovery.findAll();
+      ModelMapper.refresh(apiResources);
+
+      ModelMapper.GroupVersionKind gvk =
+          ModelMapper.getGroupVersionKindByClass(targetObj.getClass());
+      Optional<Discovery.APIResource> apiResource =
+          apiResources.stream()
+              .filter(r -> r.getKind().equals(gvk.getKind()) && r.getGroup().equals(gvk.getGroup()))
+              .findFirst();
+      if (!apiResource.isPresent()) {
+        throw new KubectlException(
+            "Cannot recognize such kubernetes resource from api-discovery: " + gvk.toString());
+      }
+      GenericKubernetesApi<KubernetesObject, KubernetesListObject> api =
+          new GenericKubernetesApi(
+              targetObj.getClass(),
+              KubernetesListObject.class,
+              gvk.getGroup(),
+              gvk.getVersion(),
+              apiResource.get().getResourcePlural(),
+              apiClient);
+      if (apiResource.get().getNamespaced()) {
+        String namespace =
+            Strings.isNullOrEmpty(targetObj.getMetadata().getNamespace())
+                ? Namespaces.NAMESPACE_DEFAULT
+                : targetObj.getMetadata().getNamespace();
+        KubernetesApiResponse<KubernetesObject> response =
+            api.create(namespace, targetObj, new CreateOptions());
+        return response.getObject();
+      } else {
+        KubernetesApiResponse<KubernetesObject> response =
+            api.create(targetObj, new CreateOptions());
+        return response.getObject();
+      }
+    } catch (Throwable t) {
```

Comment:
Catch the specific errors, not generic `Throwable`

Reply:
fixed